runtime/src/ancient_append_vecs.rs

```diff
@@ -57,7 +57,7 @@ impl<'a> AccountsToStore<'a> {
             hashes.push(account.1.account.hash);
             // we have to specify 'slot' here because we are writing to an ancient append vec and squashing slots,
             // so we need to update the previous accounts index entry for this account from 'slot' to 'ancient_slot'
-            accounts.push((&account.1.account.meta.pubkey, &account.1.account, slot));
+            accounts.push((&account.1.account, slot));
```

Comment:
this is the upstream change. Stop pushing the pubkey as a separate item in the tuple. pubkey is already available in the stored account.